source/reactor/unit_tests/EventHandlersTest.cpp

```diff
@@ -0,0 +1,92 @@
+#include <sys/socket.h>
+
+#include "EventReaderMock.h"
+#include "EventWriterMock.h"
+#include "reactor/EventLoopImpl.h"
+
+#include "gtest/gtest.h"
+
+using namespace u;
+using namespace u::reactor;
+
+using ::testing::InSequence;
+using ::testing::Invoke;
+using ::testing::StrictMock;
+
+constexpr long test_timeout = 100;
+
+namespace {
+
+TEST(EventHandlersTest, read_write_ShouldSucceed)
+{
+    EventLoopImpl loop;
+    StrictMock<EventReaderMock> reader;
+    StrictMock<EventWriterMock> writer;
```

Comment:
Is there any reason to have separate mock classes for reader and writer? We could just define all handlers in a single calls and do
```suggestion
    StrictMock<EventHandlerMock> reader;
    StrictMock<EventHandlerMock> writer;
```
Or am I missing something?

Reply:
EventLoop behaves differently depending on which handler functions have been defined. For example, if the `onWrite` function has not been defined, then the file descriptor is not added to the `fd_set` handed to the `select()` and no write event is fired.